File: app/src/main/java/com/labessence/biotech/finaldilution/component/PercentageConcentration.java
package com.labessence.biotech.finaldilution.component;

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

class PercentageConcentration extends Concentration {
    public PercentageConcentration(double amount) {
        super(amount, ConcentrationType.PERCENTAGE);
    }

    @Override
    double calcDesiredMass(double volume, double molarMass) {
        return concentration * volume / 100;
    }

    @Override
    double calcVolumeForDesiredMass(double mass, double molarMass) {
        return mass / concentration * 100;
    }
}

File: app/src/main/java/com/labwhisper/biotech/finaldilution/component/concentration/PercentageConcentration.kt
package com.labwhisper.biotech.finaldilution.component.concentration

class PercentageConcentration(amount: Double) :
    Concentration(amount, ConcentrationType.PERCENTAGE) {

    override val multiplicationFactor = 100.0

    override fun calcDesiredMass(volume: Double, molarMass: Double?): Double {
        return concentration * volume / multiplicationFactor
    }

    override fun calcVolumeForDesiredMass(mass: Double, molarMass: Double?): Double {
        return mass / concentration * multiplicationFactor
    }
}
